pkg/controller/replicator.go

```diff
@@ -354,33 +353,58 @@ type backendWrapper struct {
 	mode    types.Mode
 }
 
-func (r *replicator) RemainSnapshots() (int, error) {
-	// addReplica may call here even without any backend
-	if len(r.backends) == 0 {
-		return 1, nil
-	}
-
-	ret := math.MaxInt32
+func (r *replicator) GetSnapshotCountAndSizeUsage() (int, int64, error) {
+	countUsage := 0
+	sizeUsage := int64(0)
+	hasResult := false
 	for _, backend := range r.backends {
 		if backend.mode == types.ERR {
 			continue
 		}
-		// ignore error and try next one. We can deal with all
-		// error situation later
-		if remain, err := backend.backend.RemainSnapshots(); err == nil {
-			if remain < ret {
-				ret = remain
-			}
+		// ignore error and try next one
+		count, size, err := backend.backend.GetSnapshotCountAndSizeUsage()
+		if err != nil {
+			logrus.Errorf("Failed to get snapshot count and size usage: %v", err)
+			continue
+		}
+		hasResult = true
+		if countUsage < count {
+			countUsage = count
+		}
+		if sizeUsage < size {
+			sizeUsage = size
 		}
 	}
-	if ret == math.MaxInt32 {
-		return 0, fmt.Errorf("cannot get valid result for remain snapshot")
+
+	if !hasResult {
+		return 0, 0, fmt.Errorf("cannot get an valid result for snapshot usage")
 	}
-	if ret <= 0 {
-		return 0, fmt.Errorf("too many snapshots created")
+	return countUsage, sizeUsage, nil
+}
+
+func (r *replicator) GetHeadFileSize() (int64, error) {
+	sizeUsage := int64(0)
+	hasResult := false
+	for _, backend := range r.backends {
+		if backend.mode == types.ERR {
+			continue
+		}
+		// ignore error and try next one
+		size, err := backend.backend.GetHeadFileSize()
+		if err != nil {
+			logrus.Errorf("Failed to get head file size usage: %v", err)
+			continue
+		}
+		hasResult = true
```

Comment:
nit: ideally, should skip the loop if found the number.

Reply:
Ideally, all replicas have same data. We can just get a result from one of them and return. However, the data may not be synchronized in some cases, so I prefer to check all replicas and return biggest value.

Reply:
Knew your intention. Sounds good.